jenkinsbuilder/cccp_index_reader.py

```diff
@@ -118,23 +197,41 @@ def main(indexdlocation):
             # projectify
             with open(generated_filename, 'w') as outfile:
                 yaml.dump(project, outfile)
-
             # run jenkins job builder
             myargs = ['jenkins-jobs',
                       '--ignore-cache',
                       'update',
                       ':'.join(
                           [jjb_defaults_file, generated_filename])
                       ]
-            print myargs
-            proc = subprocess.Popen(myargs,
-                                    stdout=subprocess.PIPE)
-            proc.communicate()
+            run_command(myargs)
 
+            new_projects_names.append(
+                str(project[0]["project"]["appid"]) + "-" +
+                str(project[0]["project"]["jobid"]) + "-" +
+                str(project[0]["project"]["desired_tag"])
+            )
         finally:
-            print "Removing {}".format(t)
-            # shutil.rmtree(t)
+            pass
+    # get list of old projects
+    old_projects = get_old_project_list()
+
+    # find stale projects
+    stale_projects = find_stale_projects(
+        old_projects,
+        list(set(new_projects_names))
+    )
+
+    print stale_projects
+
+    # delete stale entries at jenkins
+    delete_stale_projects_on_jenkins(stale_projects)
+
+    # delete stale entries at openshift
+    delete_stale_projects_on_openshift()
 
+    # export the current projects_list in file
+    export_new_project_names(new_projects_names)
 
```

Comment:
that would be try catching which is already taken care in function. Doing try catch at function itself.